hw9_oop/HW_OOP.py

```diff
@@ -0,0 +1,455 @@
+Задание 1 (5 баллов)
+Напишите классы Chat, Message и User. Они должны соответствовать следующим требованиям:
+
+Chat:
+
+Должен иметь атрибут chat_history, где будут храниться все сообщения (Message) в обратном хронологическом порядке (сначала новые, затем старые)
+Должен иметь метод show_last_message, выводящий на экран информацию о последнем сообщении
+Должен иметь метод get_history_from_time_period, который принимает два опциональных аргумента (даты с которой и по какую мы ищем сообщения и выдаём их). Метод также должен возвращать объект типа Chat
+Должен иметь метод show_chat, выводящий на экран все сообщения (каждое сообщение в таком же виде как и show_last_message, но с разделителем между ними)
+Должен иметь метод recieve, который будет принимать сообщение и добавлять его в чат
+Message:
+
+Должен иметь три обязательных атрибута
+text - текст сообщения
+datetime - дата и время сообщения (встроенный модуль datetime вам в помощь). Важно! Это должна быть не дата создания сообщения, а дата его попадания в чат!
+user - информация о пользователе, который оставил сообщение (какой тип данных использовать здесь, разберётесь сами)
+Должен иметь метод show, который печатает или возвращает информацию о сообщении с необходимой информацией (дата, время, юзер, текст)
+Должен иметь метод send, который будет отправлять сообщение в чат
+User:
+
+Класс с информацией о юзере, наполнение для этого класса придумайте сами
+Напишите несколько примеров использования кода, которое показывает взаимодействие между объектами.
+
+В тексте задания намерено не указано, какие аргументы должны принимать методы, пускай вам в этом поможет здравый смысл)
+
+В этом задании не стоит флексить всякими продвинутыми штуками, для этого есть последующие
+
+В этом задании можно использовать только модуль datetime
+
+from datetime import datetime
+import time
+​
+class Chat:
+    def __init__(self):
+        self.chat_history = []
+        
+    def show_last_message(self):
+        last_message = self.chat_history[0]
+        last_message.show()
+    
+    def get_history_from_time_period(self, after = datetime(2000, 1, 1, 1, 1, 1),
+                                     before = datetime(2099, 1, 1, 1, 1, 1)):
```

Comment:
Лучше не хардкодить, а сделать None по-умолчанию. Уже в коде можно это обработать и взять дэйттайм первого и последнего сообщений
```suggestion
    def get_history_from_time_period(self, after=None, before=None):
```